### ISSUE TITLE
Add support for creating tasks to format `.gradle.kts` files

### ISSUE THREAD

**Issue by cortinico** [cortinico/ktfmt-gradle#337]
## ⚠️ Is your feature request related to a problem? Please describe

We should create tasks that format the gradle.kts files (like `ktfmtFormatKotlinScript` and `ktfmtCheckKotlinScript`) and hook them inside the `ktfmtFormat` and `ktfmtCheck` tasks.

## 💡 Describe the solution you'd like
Could be implemented in a number of ways. Happy to discuss any approach

## 🤚 Do you want to develop this feature yourself?
- [ ] Yes
- [X] No

This is up for grab :)

**Comment by simonhauck** [comment]
I think this is a great idea.
One easy way to implement this, would be to take the project directory and only collect the top level .kts. files. That would probably solve most use cases.

If we searched recursively for all .kts files, it would be a bit weird for nested multi-module projects. Because the parent project would also format the gradle scripts of subprojects. What do you think?

**Comment by cortinico** [comment]
> If we searched recursively for all .kts files, it would be a bit weird for nested multi-module projects. Because the parent project would also format the gradle scripts of subprojects. What do you think?

Yeah that's a good point, so we get the top level `.gradle.kts` and `.settings.kts` file and we're good to go 👍 Each project will inspect its own kts file

**Comment by jessestricker** [comment]
For an ad-hoc solution, I've implented this in my `build.gradle.kts` like the following and it works really well for now:

```kotlin
val scriptFiles = fileTree(projectDir) { include("*.kts") }

val ktfmtFormatScript by
    tasks.registering(KtfmtFormatTask::class) {
        description = "Run Ktfmt formatter for script files on project '${project.name}'."
        source = scriptFiles
    }

tasks.ktfmtFormat { dependsOn(ktfmtFormatScript) }

val ktfmtCheckScript by
    tasks.registering(KtfmtCheckTask::class) {
        description =
            "Run Ktfmt formatter validation for script files on project '${project.name}'."
        source = scriptFiles
    }

tasks.ktfmtCheck { dependsOn(ktfmtCheckScript) }
```

**Comment by cortinico** [comment]
> For an ad-hoc solution, I've implented this in my `build.gradle.kts` like the following and it works really well for now:

I know but I think we can add support for `ktfmtFormatScript` and `ktfmtCheckScript` out of the box for ktfmt-gradle
